File: src/VacanciesService/VacanciesService.Domain/Models/Vacancy.cs
namespace VacanciesService.Domain.Models
{
    public class Vacancy
    {
        public Guid Id { get; set; }
        public DateTime CreatedAt { get; set; }
        public string Title { get; set; }
        public string EmploymentType { get; set; }
        public Guid CompanyId { get; set; }
        public bool Archived { get; set; }
        public DateTime DeadlineAt { get; set; }
        public VacancyDetails VacancyDetails { get; set; }
    }
}

File: frontend/src/app/core/models/vacancies/vacancy.ts
import VacancyDetails from './vacancyDetails';

export default interface Vacancy {
  id: string;
  companyId: string;
  title: string;
  employmentType: string;
  archived: boolean;
  createdAt: Date;
  deadlineAt: Date;
  vacancyDetails?: VacancyDetails;
}
